services/web/server/src/simcore_service_webserver/projects/db.py

```diff
@@ -444,6 +446,54 @@ async def get_project(
                 project_type,
             )
 
+    async def check_project_node_has_all_required_inputs(
+        self, user_id: UserID, project_uuid: ProjectID, node_id: NodeID
+    ) -> None:
+        project_dict, _ = await self.get_project(user_id, f"{project_uuid}")
```

Comment:
If I may be candid, it seems like the controller/service/repository design pattern isn't being fully adhered to here (which we are trying to comply with). All of this logic should, in my opinion, be in the service layer (projects_api.py) as a function. You can even name it the same, check_project_node_has_all_required_inputs, as this is specific business logic. In db.py, you should have a pure, simple interface that gets data from the database, and that's it. From what I can see, you don't even need to define a new function here; just use the `get_project` function or `get_project_db`, and that's it.

Reply:
I agree with matus

Reply:
also agree. the webserver is fat enough, please do not add logic inside the db layer thanks, also if I need to re-use this at some point for the computational services I would vastly prefer to have it in the api.